python/cudf/cudf/tests/test_column.py

```diff
@@ -29,8 +29,37 @@
 
 @pytest.fixture(params=dtypes, ids=dtypes)
 def pandas_input(request):
-    data = np.random.randint(0, 1000, 100)
-    return pd.Series(data, dtype=request.param)
+    dtype = request.param
+    rng = np.random.default_rng()
+    size = 100
+
+    def random_ints(dtype, size):
+        dtype_min = np.iinfo(dtype).min
+        dtype_max = np.iinfo(dtype).max
+        return rng.integers(dtype_min, dtype_max, size=size, dtype=dtype)
+
+    try:
+        dtype = np.dtype(dtype)
+    except TypeError:
+        if dtype == "category":
+            data = random_ints(np.int64, size)
+        else:
+            raise
+    else:
+        if dtype.kind == "b":
+            data = rng.choice([False, True], size=size)
+        elif dtype.kind in ("m", "M"):
+            # datetime or timedelta
+            data = random_ints(np.int64, size)
+        elif dtype.kind == "U":
+            # Unicode strings of integers like "12345"
+            data = random_ints(np.int64, size).astype(dtype.str)
+        elif dtype.kind == "f":
+            # floats in [0.0, 1.0)
+            data = rng.random(size=size, dtype=dtype)
+        else:
+            data = random_ints(dtype, size)
```

Comment:
Seems like this could be more generally useful. Which also makes me wonder... does something like this already exist in cudf, and can we re-use that?

Reply:
That's definitely a good question. I plan to investigate that further and strive for de-duplication as I work through some more warnings. Lots of the warnings are affected by dtypes!